### ISSUE TITLE
Support disabling watch mode

### ISSUE THREAD

**Issue by jimmyzzxhlh** [jest-community/vscode-jest#613]
There doesn't seem to be a way to disable the watch mode. With watch mode on the tests are always running whenever there is a save action on any file in the workspace. When we have a decent number of tests in the workspace it takes some time to finish all tests, which causes a high cpu load on my machine (observed by just listening to the fan noise).

Is it possible to disable the watch mode so that we have the ability to control when the tests are being run?

**Comment by connectdotz** [comment]
interesting use case... just to make sure we are on the same page. 

Are you concerning the initial full test run or the subsequent watch-mode run? The initial full test run can be turned off but not the watch-mode right now... In theory, the watch mode should only run the changed tests, which should be a smaller scope than the full run. Are you asking to turn off full and watch-mode all together and just have an "on-demand full test run" instead?

**Comment by jimmyzzxhlh** [comment]
@connectdotz I'm more concerning about the subsequent watch-mode run. The initial full test run is OK since it's a one-time thing and it can be turned off. However for subsequent runs the problem is the cpu load will spike up every time I save something so it becomes kind of annoying. An on-demand full test run seems to fit my need.

**Comment by ifiokjr** [comment]
@connectdotz This is essential in my opinion. 

Once the number of tests exceeds 100 it becomes impossible to use this plugin without the constant need to restart VS Code due to the high CPU load. Switched on, in a sufficiently large project, I'm restarting VS Code once or twice every hour.

As a result my current workflow is this.

- Identify a broken test
- Re-enable this plugin for the workspace. 
- Debug the broken test.
- Disable this plugin for the workspace. 

Watch mode as it currently is fails when working on any core file. An edit causes 90% of the tests to run.

For a real life example of this: 

- Clone https://github.com/remirror/remirror 
	```bash 
	git clone https://github.com/remirror/remirror
	```

- Install dependencies with `pnpm i` - **pnpm** can be installed via `npm i -g pnpm`.
- Open this file to edit `packages/@remirror/core-helpers/src/core-helpers.ts`
- Add a few edits and save maybe 5 or 6 times.
- Listen to the harmonious symphony of your machine's overworked fan.

For full disclosure I'm seeing these problems on a fully specced `64GB` 2017 `iMac`. I had actually stopped using the plugin at all, for the past year, but I love how easy this extension makes it to debug tests, and I appreciate how good this plugin is so many features.

**Comment by connectdotz** [comment]
@ifiokjr thanks for the test repo. I agree it will be nice to have an on-demand mode... 

@jimmyzzxhlh and @ifiokjr, please let me know if the following workflow works for your use cases:

- enable `jest.onDemand` or something like this in your config
- when you open the project, the plugin will not run any test automatically 
- when you are ready to run tests, you can use the Command Palette (or maybe configure keyboard shortcut) to trigger jest full run (or your custom `jest.pathToJest` if configured)

**Comment by jimmyzzxhlh** [comment]
@connectdotz Yes this works for me.

**Comment by MatheusPoliCamilo** [comment]
@jimmyzzxhlh how did you do this "onDemand" configuration?

**Comment by jimmyzzxhlh** [comment]
> @jimmyzzxhlh how did you do this "onDemand" configuration?

That doesn't exist, and it's exactly what I'm asking for here :)

**Comment by MatheusPoliCamilo** [comment]
Ah ok, you said it worked, I thought you found something hahah thanks!

**Comment by ematta** [comment]
I tried to disabled:

```
"jest.autoEnable": false,
"jest.runAllTestsFirst": false,
```

and watchmode is still going crazy. It's making this plugin nearly unusable.

**Comment by rnewman** [comment]
I ended up turning off this plugin for this reason. Running tests in the Terminal isn't so bad…

**Comment by dave-vazquez** [comment]
Yeah, I've only been using this a day and this extension is very CPU intensive. I may have to uninstall this if there's no option to disable watch-mode sometime soon. Looking forward to this option being added.

**Comment by sldsrg** [comment]
Strange that ``"jest.autoEnable": false,`` turns watch mode **off**, but ``"jest.runAllTestsFirst": false,`` ignores previous parameter and turns watch mode back **on**.

**Comment by connectdotz** [comment]
Hi, we have addressed this issue in [v4.0.0-alpha.5](https://github.com/jest-community/vscode-jest/releases/tag/v4.0.0-alpha.5), please feel free to give it a try and let us know if it works for you.
